projects/desktop/src/interfaces/generics.d.ts

```diff
@@ -0,0 +1,20 @@
+/*
+  File containing generics typescript interface for type checking
+*/
+
+interface IStringTMap<T> {
```

Comment:
What does it mean ?

Reply:
With this you can make an object with string keys and a T type for the value. Example:
`
    const myObject: IStringTMap<Number> = {'one' : 1}
    const usersDetails: IStringTMap<UserDetail> = {'username': {...}}
`

Reply:
Okay I see and why `**[**key: string**]**` ?

Reply:
The crochets

Reply:
It is used to isolate the type declaration